modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj

```diff
@@ -559,9 +561,14 @@
 (doseq [op [:= :!= :< :<= :> :>= :between]]
   (defmethod sql.qp/->honeysql [:sqlserver op]
     [driver [_ field :as clause]]
-    (binding [*compared-field-options* (when (and (vector? field)
-                                                  (= (get field 0) :field))
-                                         (get field 2))]
+    (binding [*compared-field-options*
+              (when (and (vector? field)
+                         (= (get field 0) :field))
```

Comment:
Is there a reason not to use more succinct `(field 0)` version?

Reply:
Yes, I don't want this test to blow up if something unexpected comes in. In fact, I'll replace the `(field 1)` in the next line too.

Reply:
I found myself preferring using `get` too, in that order: if it's a literal keyword, use it (`(:a m)`), if it's a literal map (or at least some global static thingie or something defined right in place), use it (`({1 2} 1)`), in other cases, use `get` to get at least somewhat understandable exception when things go wrong.

Reply:
This particular case is about vectors. I use the function call notation when I want to get an index out of bounds exception if the index is wrong. Here, I want the test to fail silently, which happens when `get` returns `nil`.